### DISCUSSION TITLE
Personal lifetime license

### DISCUSSION THREAD

**Discussion by izhangzhihao** [izhangzhihao/intellij-rainbow-brackets#2587] [General]
**🛒 Get your lifetime-license**  
Purchase at [Buy Me a Coffee](https://buymeacoffee.com/rainbowbrackets/e/303043):  
→ Receive JetBrains license **within 3 minutes**  
→ Missing your license? Contact: rainbowbrackets.plugin@gmail.com

**⚡ Lifetime Access Guarantee**  
✅ **Auto-renewed before expiration** (bot handled)  
✅ **Zero manual effort**, uninterrupted usage  
> 📌 Note: "Lifetime" = automated renewal model

Changed your email or entered an incorrect one? Link your incorrect address to the correct email [here](https://sales.jetbrains.com/hc/en-gb/articles/7654171906706-Link-multiple-email-addresses-to-your-JetBrains-Account)

**🛒 获取许可证**  
通过 [这里](https://docs.qq.com/doc/DQ3NaT1NJVUJka2h0) 购买：  
→ **首次许可需要手动触发注册，可能会有一些延迟** 
→ 如果没收到许可请联系：rainbowbrackets.plugin@gmail.com

**⚡ 永久使用权保障**  
✅ **到期前自动续期**（自动执行）  
✅ **零手动操作**，持续使用无中断  
> 📌 注："lifetime license" == 自动续期模式

如果已更改邮箱或预留了错误邮箱，你可以将错误邮箱地址关联到正确的邮箱。请参考 [here](https://sales.jetbrains.com/hc/zh-cn/articles/7654171906706-将多个电子邮件地址关联到您的-JetBrains-Account-下)

如果你想购买国区年付许可，请参考[这里](https://docs.qq.com/doc/DQ3NaT1NJVUJka2h0)。

# How to use the discount code

open https://www.jetbrains.com/store/redeem/?plugins=&product=PRAINBOWBRACKET 

<img width="940" alt="image" src="https://github.com/user-attachments/assets/c8beb1ed-4d91-484a-bd0e-ad7efd336293">

**Comment by izhangzhihao** [comment]
GitHub  Action Workflow completed: automatically sending renewal email to lifetime license users(with existing licenses expiring next month). 🎉

**Reply by fanjiaqi1995** [reply]
我平时用的公司电脑的JetBrains许可证。离职后我还能给自己电脑自己的账户用吗

**Reply by izhangzhihao** [reply]
> 我平时用的公司电脑的JetBrains许可证。离职后我还能给自己电脑自己的账户用吗

如果你使用的是公司的 JetBrains 账户，离职后一般来讲你将无法访问公司的 JetBrains 账户，也将无法继续使用许可。
建议你使用个人邮箱注册 JetBrains 账户，然后激活许可，同时通过个人 JetBrains 账户的[离线方式激活](https://sales.jetbrains.com/hc/en-gb/articles/360016995379-Activating-JetBrains-IDEs-with-an-offline-activation-code)，这样同一个 IDE 可以同时使用公司和个人的账户。
如果你已经使用了公司邮箱激活许可，可以在下一年的周期中换到个人账户上激活新的许可。

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sending renewal email to lifetime license users(with existing licenses expiring next month). 🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sending renewal email to lifetime license users(with existing licenses expiring next month). 🎉

**Comment by Mick235711** [comment]
已支付，支付宝留言区写不下了（`winmikedows@hotmail.com`）

**Reply by izhangzhihao** [reply]
一封邮件会自动发出，请查收，如果没收到也可以检查垃圾箱，再次感谢支持。

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sending renewal email to lifetime license users(with existing licenses expiring next month). 🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sending renewal email to lifetime license users(with existing licenses expiring next month). 🎉

**Comment by dantetwc** [comment]
not yet received license waichuntwc@gmail.com

**Reply by izhangzhihao** [reply]
pls check your trash box, it has already been sent to you.

**Reply by dantetwc** [reply]
Got it thanks

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sending renewal email to lifetime license users(with existing licenses expiring next month). 🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sending renewal email to lifetime license users(with existing licenses expiring next month). 🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sending renewal email to lifetime license users(with existing licenses expiring next month). 🎉

**Comment by Ender-Jones** [comment]
enderjones1872@gmail.com
zfb的永久许可,谢谢

**Reply by izhangzhihao** [reply]
昨天就已经发了，大概在垃圾收件箱？ 麻烦帮忙点一下"这不是垃圾邮件"

**Reply by Ender-Jones** [reply]
谢谢,找到了,已经mark了不是垃圾邮件(气晕)

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sending renewal email to lifetime license users(with existing licenses expiring next month). 🎉

**Comment by lipangeng** [comment]
lipg@outlook.com
23年6月7日微信支付的。并未收到新的折扣码，垃圾邮件也检查没有。请再次发送，谢谢。

**Reply by izhangzhihao** [reply]
五月发出了，可能垃圾邮件过期被清理了，我又发了一遍，请查收

**Reply by lipangeng** [reply]
嗯，这次看到了，感谢。应该是垃圾箱过期被清理了，那时候还没有加白名单。

**Comment by lipangeng** [comment]
嗯，这次看到了，感谢。应该是垃圾箱过期被清理了，那时候还没有加白名单。

2024年6月14日 14:57，Zhihao Zhang ***@***.***> 写道：



五月发出了，可能垃圾邮件过期被清理了，我又发了一遍，请查收

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sending renewal email to lifetime license users(with existing licenses expiring next month). 🎉

**Comment by casper-yj** [comment]
yanjia2304@126.com
wx永久许可，谢谢

**Reply by izhangzhihao** [reply]
已经发了，查收一下收件箱/垃圾箱呢

**Reply by casper-yj** [reply]
收到，谢谢

**Comment by PanJianxin** [comment]
239692275@qq.com
支付宝付款的永久许可，可能被垃圾邮件自动清理了，能再发一次吗？

**Reply by izhangzhihao** [reply]
发了，检查下垃圾收件箱，并加为联系人哈～

**Reply by PanJianxin** [reply]
OK，收到了，谢谢

**Comment by Fire1nRain** [comment]
gjx8010@hotmail.com
支付宝付款的永久许可，谢谢~

**Reply by izhangzhihao** [reply]
已发送，请查收

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sending renewal email to lifetime license users(with existing licenses expiring next month). 🎉

**Comment by TheWhiteDog9487** [comment]
想问一下教育优惠是持续多长时间呢？在校期间都能续签吗还是什么？
另外我的学校拥有但拒绝给学生提供教育邮箱，我能拿出学信网证明，学校教务系统截图等证明我的身份，这种情况下能申请到教育优惠吗？

**Reply by izhangzhihao** [reply]
邮件我呗，内容真实就没问题

**Comment by ant5033** [comment]
I received my first license on August 4, 2023, and the current date is August 9, 2024, but I still haven't received the email with the discount code. I've checked both the trash and spam folders, but I couldn't find it. Please help me resolve this issue. I sent an inquiry email with my previous license information to ant5033@naver.com.

**Reply by izhangzhihao** [reply]
already replied to your email, pls check your trash-box or inbox

**Reply by ant5033** [reply]
Thank you. I was about to renew using the email I received, but it turns out it was automatically renewed! I appreciate your prompt response.

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sending renewal email to lifetime license users(with existing licenses expiring next month). 🎉

**Comment by ltto** [comment]
支付宝 已支付
jb邮箱121172722@qq.com

**Reply by izhangzhihao** [reply]
已发送，请查收（可能在垃圾收件箱）

**Comment by SmushyTaco** [comment]
I purchased this through the "Buy Me a Coffee" option with the email "personal@nikanradan.com" but I've received no email with a license code.

**Reply by izhangzhihao** [reply]
Sorry for the late response. The issue has been resolved. Please check your email.

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sending renewal email to lifetime license users(with existing licenses expiring next month). 🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sending renewal email to lifetime license users(with existing licenses expiring next month). 🎉

**Comment by jcy9503** [comment]
I purchased lifetime license. Thanks for your effort.
jcy9503@naver.com

**Reply by izhangzhihao** [reply]
Here is your license details

2024-11-02T06:42:09.4433525Z License Type: PERSONAL
2024-11-02T06:42:09.4434660Z Trial: False
2024-11-02T06:42:09.4435131Z Period: Annual
2024-11-02T06:42:09.4435681Z Start From: 2024-06-15
2024-11-02T06:42:09.4436536Z Valid Till: 2025-11-01
2024-11-02T06:42:09.4437142Z Overuse Till: 2025-11-08
2024-11-02T06:42:09.4437685Z Owner Code: 8698063
2024-11-02T06:42:09.4438499Z Owner Type: Personal
2024-11-02T06:42:09.4439029Z Owner Email: jcy9503@naver.com

**Comment by azeek** [comment]
I have purchased a lifetime license. Thank you.
My JetBrains email : zkdlfhrm@gmail.com

**Reply by izhangzhihao** [reply]
Here's your license details 
{"licenses":[{"type":"PERSONAL","trial":false,"period":"Annual","startFrom":[2024,11,14],"validTill":[2025,11,15],"overuseTill":[2025,11,22],"ownerCode":1075104,"ownerType":"Personal","ownerEmail":"zkdlfhrm@gmail.com"}]}

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sending renewal email to lifetime license users(with existing licenses expiring next month). 🎉

**Comment by mazenmohamed3** [comment]
mazenmohamed.mm956@gmail.com

hello i subscribed for lifetime on this account

**Reply by izhangzhihao** [reply]
Hi,this is your license details


```
2024-12-08T12:06:26.0212683Z License Type: PERSONAL
2024-12-08T12:06:26.0213315Z Trial: False
2024-12-08T12:06:26.0213946Z Period: Annual
2024-12-08T12:06:26.0214559Z Start From: 2024-12-08
2024-12-08T12:06:26.0215280Z Valid Till: 2025-12-07
2024-12-08T12:06:26.0216530Z Overuse Till: 2025-12-14
2024-12-08T12:06:26.0217145Z Owner Code: 21354545
2024-12-08T12:06:26.0217922Z Owner Type: Personal
2024-12-08T12:06:26.0218520Z Owner Email: mazenmohamed.mm956@gmail.com
2024-12-08T12:06:26.0218994Z 
```

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sent renewal emails to lifetime license users or processed license renewals via the JetBrains API.🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sent renewal emails to lifetime license users or processed license renewals via the JetBrains API.🎉

**Comment by Daschi1** [comment]
@izhangzhihao It seems like this might now be possible with "perpetual licenses," which have been introduced by JetBrains for the marketplace: https://blog.jetbrains.com/platform/2025/01/introducing-perpetual-licenses-on-jetbrains-marketplace

**Reply by izhangzhihao** [reply]
Yes, JetBrains now offers a perpetual license type, but this type of license was discussed with me before its official release (probably about a year ago). At that time, I clearly pointed out that this license type has a very dangerous flaw. I contacted JetBrains via email, and they acknowledged receiving the email but made no changes. Therefore, until JetBrains addresses this issue, I will not support this feature.

**Reply by Daschi1** [reply]
I see, thanks for sharing your perspective. Would you mind elaborating on the flaw you mentioned? Appreciate your insights!

**Reply by luckydonald** [reply]
@izhangzhihao could you explain the dangerous flaw, please? Would love to understand.

**Comment by jlu248** [comment]
兄弟，支付宝买的一年
邮箱：jasonlu0248@gmail.com

**Reply by izhangzhihao** [reply]
=add context file in master branch☑ Done✅ done

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sent renewal emails to lifetime license users or processed license renewals via the JetBrains API.🎉

**Comment by amiPark** [comment]
I purchased your good program.
arms1146@gmail.com

**Reply by izhangzhihao** [reply]
done, pls check your mail box

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sent renewal emails to lifetime license users or processed license renewals via the JetBrains API.🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sent renewal emails to lifetime license users or processed license renewals via the JetBrains API.🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sent renewal emails to lifetime license users or processed license renewals via the JetBrains API.🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sent renewal emails to lifetime license users or processed license renewals via the JetBrains API.🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sent renewal emails to lifetime license users or processed license renewals via the JetBrains API.🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: automatically sent renewal emails to lifetime license users or processed license renewals via the JetBrains API.🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: Sent license renewal emails and processed renewals via the JetBrains API. 🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: Sent license renewal emails and processed renewals via the JetBrains API. 🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: Sent license renewal emails and processed renewals via the JetBrains API. 🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: Sent license renewal emails and processed renewals via the JetBrains API. 🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖[Bot] GitHub Action completed: Sent license renewal emails and processed renewals via the JetBrains API. 🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖 Auto renew completed: Sent renewal emails and processed renewals via the JetBrains API. 🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖 Auto renew completed: Sent renewal emails and processed renewals via the JetBrains API. 🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖 Auto renew completed: Sent renewal emails and processed renewals via the JetBrains API. 🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖 Auto renew completed: Sent renewal emails and processed renewals via the JetBrains API. 🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖 Auto renew completed: Sent renewal emails and processed renewals via the JetBrains API. 🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖 Auto renew completed: Sent renewal emails and processed renewals via the JetBrains API. 🎉

**Comment by intellij-rainbow-brackets** [comment]
🤖 Auto renew completed: Sent renewal emails and processed renewals via the JetBrains API. 🎉
